### DISCUSSION TITLE
tutorials/assets_pipeline/exporting_3d_scenes

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#187] [User-contributed notes]
# tutorials/assets_pipeline/exporting_3d_scenes

Overview: In Godot, it is possible to export 3D scenes as a glTF 2.0 file. You can export as a glTF binary (.glb file) or glTF embedded with textures ( gltf+.bin+ textures). This allows you to crea...

https://docs.godotengine.org/en/stable/tutorials/assets_pipeline/exporting_3d_scenes.html

**Comment by llama-nl** [comment]
How to do this using Code?

**Reply by sh4wn** [reply]
I need save some node to stl/obj file in runtime, is there any way to do so?

**Reply by Calinou** [reply]
> How to do this using Code?

This is described by the page linked at the bottom: [Runtime file loading and saving](https://docs.godotengine.org/en/stable/tutorials/io/runtime_file_loading_and_saving.html#doc-runtime-file-loading-and-saving-3d-scenes)

> I need save some node to stl/obj file in runtime, is there any way to do so?

Godot doesn't include a class that writes STL/OBJ files, but an OBJ add-on for this exists: https://github.com/fbcosentino/godot-objexporter

**Comment by ghost** [comment]
Video about process of exporting CSG to Blender and back to Godot https://www.youtube.com/watch?v=gIK02BgCCBI
